db/sovereign-db/src/state_db.rs

```diff
@@ -97,10 +128,12 @@ impl TreeReader for StateDB {
 impl TreeWriter for StateDB {
     fn write_node_batch(&self, node_batch: &jmt::storage::NodeBatch) -> anyhow::Result<()> {
         for (node_key, node) in node_batch.nodes() {
+            assert_eq!(node_key.version(), self.get_next_version());
             self.db.put::<JmtNodes>(node_key, node)?;
         }
 
         for ((version, key_hash), value) in node_batch.values() {
+            assert_eq!(version, &self.get_next_version());
```

Comment:
Should we pull the `get_next_version()` call out into a variable so that we're not locking and unlocking the mutex in a loop? Use something like `let next_version = self.get_next_version();` outside of the loop, and then just `assert_eq!(version, next_version);` in each iteration.

If we do, I guess we should also handle the failure case since the local copy of `next_version` could be stale.
```rust
if version != next_version { 
  next_version = self.get_next_version;
  assert_eq!(version, next_version);
}
```

Reply:
Synced with @bkolad offline. We removed version checks from the TreeWriter entirely in favor of the new `update_db` method. Resolving this conversation.